---
float32 x_pos
float32 y_pos
float32 z_pos